### ISSUE TITLE
Crashing When Undo'ing Too Much

### ISSUE THREAD

**Issue by MrTeeXD** [olive-editor/olive#720]
### System Information

**Olive version:**  [f47a7dad]
**Source:** [zip from github]
**Operating system:** [Windows 10 64-bit]
**CPU:** [Intel i5-6400]
**RAM:** [4GB]
**GPU:** [Intel Graphics HD 530 128MB (Latest Version)]

### Detailed Description

[When the user undo's too much, the program decides to crash]

### Steps to reproduce

[Press Ctrl + Z alot]

### Output Log

```
[Nothing]
```

### GDB/CDB Backtrace (for crashes)

```
[Nothing]
```

**Comment by itsmattkc** [comment]
You'll need to provide specifics on what you were undoing. Undoing "too much" is not the root cause here.

**Comment by MisterTeeXD** [comment]
Alright, i missed placed a image in my timeline, then i wanted to undo it and i may have did that a little too much

**Comment by itsmattkc** [comment]
That's still not very specific I'm afraid. Can you recreate it consistently and provide specific steps?

**Comment by MrTeeXD** [comment]
Let me try again, so, i placed an image on the timeline and i selected the image to Keyframe it, so i activated the Keyframe and moved the image off the screen, when the playhead has touched the end of the image, i though something was broken and could not moved the image, so i pressed Ctrl + Z to fix it but i pressed it too much and in result, crashed the program

**Comment by itsmattkc** [comment]
Can you recreate it, as in are you able to trigger the same crash again?

**Comment by MrTeeXD** [comment]
Yes, it should be fairly easy to trigger it again at a rapid motion of the key presses

**Comment by itsmattkc** [comment]
Have you been able to do so?

**Comment by MrTeeXD** [comment]
Yes, it crashes when undo'ing twice now

**Comment by DaniSeeh** [comment]
Could you do a Backtrace on the crash? It would surely help solve it. Or perhaps send a video of what you did to reproduce the crash?

https://github.com/olive-editor/olive/wiki/How-to-get-output-logs-if-the-program-crashes-and-use-GDB-LLDB-for-more-detailed-logs

**Comment by itsmattkc** [comment]
Yes I think a backtrace will be necessary here. Unfortunately on Windows this is not trivial since both GDB and CDB are not easily acquirable there. This may be a lost cause unless someone else can recreate this and provide a backtrace.

**Comment by itsmattkc** [comment]
I don't think this can be solved without a GDB backtrace I'm afraid.
